File: backend/QuanLyDiem/QuanLyDiem.API/Models/Student.cs
﻿using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace QuanLyDiem.API.Models
{
    public class Student
    {
        [Key]
        public int StudentId { get; set; }

        [Required(ErrorMessage = "Mã sinh viên không được để trống")]
        [RegularExpression(@"^\d{10}$", ErrorMessage = "Mã sinh viên phải bao gồm chính xác 10 chữ")]
        public string StudentCode { get; set; }

        [Required(ErrorMessage = "Họ và tên lót không được để trống.")]
        public string LastName { get; set; } = null!;

        [Required(ErrorMessage = "Tên sinh viên không được để trống.")]
        public string FirstName { get; set; } = null!;

        [Required(ErrorMessage = "Vui lòng chọn giới tính.")]
        public string Gender { get; set; } = null!;

        [Required(ErrorMessage = "Ngày sinh là bắt buộc")]
        [DataType(DataType.Date)]
        public DateTime DateOfBirth { get; set; }

        [Required(ErrorMessage = "Vui lòng chọn lớp sinh hoạt.")]
        public int HomeroomClassId { get; set; }

        [ForeignKey("HomeroomClassId")]
        public HomeroomClass? HomeroomClass { get; set; }

        [EmailAddress(ErrorMessage = "Email không đúng định dạng")]
        [Required(ErrorMessage = "Email là bắt buộc")]
        public string Email { get; set; }

        public ICollection<Enrollment>? Enrollments { get; set; } = new List<Enrollment>();
    }
}

File: frontend/client-app/src/app/models/student.ts
// Model dùng để hứng dữ liệu trả về từ API (.NET trả về)
export interface StudentResponse {
  studentId: number;
  studentCode: string;
  lastName: string;
  firstName: string;
  gender: string;
  dateOfBirth: string; // Nhận về dạng chuỗi ISO từ .NET để dễ hiển thị và format
  homeroomClassId: number;
  homeroomClassName: string;
  email: string;
}

// Model dùng để gửi dữ liệu từ Form lên API khi Thêm mới hoặc Sửa
export interface StudentCreateUpdate {
  studentCode: string;
  lastName: string;
  firstName: string;
  gender: string;
  dateOfBirth: string; // Gửi lên dạng chuỗi YYYY-MM-DD từ HTML input date
  homeroomClassId: number;
  email: string;
}